src/Test/Hspec/Expectations.hs

```diff
@@ -88,6 +89,23 @@ with_loc(shouldBe, (Show a, Eq a) => a -> a -> Expectation)
 actual `shouldBe` expected = expectTrue ("expected: " ++ show expected ++ "\n but got: " ++ show actual) (actual == expected)
 
 -- |
+-- @actual \`shouldBeNear\` expected@ sets the expectation that @actual@ be a
+-- floating point value near @expected@. If either value is zero we check that
+-- the absolute difference is less than epsilon (1e-15) otherwise we check if
+-- the relative difference is less than epsilon.
+with_loc(shouldBeNear, (Show a, Ord a, Floating a) => a -> a -> Expectation)
+actual `shouldBeNear` expected
+    -- Short circuit if they are actually equal.
+    | actual == expected = reportFail (actual == expected)
+    | actual == 0 || expected == 0 = reportFail (absDifference < epsilon)
+    | otherwise =  reportFail (relativeDifference < epsilon)
+    where
+        epsilon = 1e-15
+        absDifference = abs (actual - expected)
```

Comment:
for me it's much easier to work with code that does not use abbreviations, the reason for this is that I have to use speech recognition for programming due to RSI. for that reason I would much prefer `absoluteDifference` here